flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierTrackerTest.java

```diff
@@ -607,13 +606,26 @@ public void testTwoLastBarriersOneByOne() throws Exception {
         ValidatingCheckpointHandler validator = new ValidatingCheckpointHandler();
         inputGate = createCheckpointedInputGate(2, sequence, validator);
 
-        for (BufferOrEvent boe : sequence) {
-            assertEquals(boe, inputGate.pollNext().get());
-            Thread.sleep(10);
-        }
+        // start checkpoint 1
+        assertEquals(sequence[0], inputGate.pollNext().get());
+        Thread.sleep(10);
+
+        // start checkpoint 2
+        long start = System.currentTimeMillis();
+        assertEquals(sequence[1], inputGate.pollNext().get());
+        Thread.sleep(1);
```

Comment:
nit: maybe keep it as sleep(10)?

Do you know whether the alignment timer is the same as the system timer?

If not, the test may still be unstable until we introduce the manual clock?

Reply:
We don't need the high sleep time here. I use 1 ms as protection against zero alignment time. So I just want to have an alignment time of more than zero and 1 ms is enough